src/accountManagement/accountTransactionsFactory.ts

```diff
@@ -67,39 +72,48 @@ export class AccountTransactionsFactory {
             Buffer.from(options.serviceID).toString("hex"),
         ];
 
-        return new TransactionBuilder({
-            config: this.config,
-            sender: sender,
+        const transaction = new Transaction({
+            sender,
             receiver: sender,
-            dataParts: dataParts,
-            gasLimit: this.config.gasLimitSetGuardian,
-            addDataMovementGas: true,
-        }).build();
+            chainID: this.config.chainID,
+            gasLimit: 0n,
+        });
+
+        this.setTransactionPayload(transaction, dataParts);
+        this.setGasLimit(transaction, undefined, this.config.gasLimitSetGuardian);
+
+        return transaction;
     }
 
     createTransactionForGuardingAccount(sender: Address): Transaction {
         const dataParts = ["GuardAccount"];
 
-        return new TransactionBuilder({
-            config: this.config,
-            sender: sender,
+        const transaction = new Transaction({
```

Comment:
In a future major version, perhaps we can receive the parameters of the method as an "options" object, and allow the user to provide a `gasLimit` if she really wants to.

Reply:
Oh, sorry, not applicable anymore?